### PR TITLE
Makman2/not linters anymore

### PR DESCRIPTION
**Pull request by Makman2**

### CODE REVIEW

**Comment by sils1297** [comment]
@Makman2 can you correct commit messages? Those are not refactored to the new linter API...

**Comment by sils1297** [comment]
and rebase since I'm merging https://github.com/coala-analyzer/coala-bears/pull/393

File: bears/python/PyCommentedCodeBear.py [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-    def lint(self, filename, file):
-        output = list(eradicate.filter_commented_out_code(''.join(file)))
-        return self.process_output(output, filename, file)
-
     def run(self, filename, file):
         """
         Detects commented out source code in Python.
         """
-        return self.lint(filename, file)
+        corrected = list(eradicate.filter_commented_out_code(''.join(file)))
```
Comment (sils1297) [inline bears/python/PyCommentedCodeBear.py]: why not tuple?

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/392/commits/629cf48869a4313c58ce275021726eab931081b0 https://github.com/coala-analyzer/coala-bears/pull/392/commits/21e80571dc43d106931602b57051fb122f172ff2#diff-85cbdb9ffc9d72ae7e6a1b21154ee82fR31

File: bears/js/JSONFormatBear.py [hunk 2]
```diff
@@ -48,8 +26,29 @@ def run(self,
...
+
+        corrected = json.dumps(json_content,
+                               sort_keys=json_sort,
+                               indent=tab_width,
+                               ensure_ascii=not keep_unicode).splitlines(True)
+        # Because of a bug in several python versions we have to correct
+        # whitespace here.
+        corrected = [line.rstrip(" \n") + "\n" for line in corrected]
+
+        diffs = Diff.from_string_arrays(file, corrected).split_diff()
```
Comment (sils1297) [inline bears/js/JSONFormatBear.py]: let's not split this one, if keys are sorted, applying only one of the splitted diffs may result in wrong stuff. Be sure to use the ranges of the splitted diff though

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/392/commits/7c4b2a9cc06c61ecb4008d3ab3f70da6ee86b8bb

File: bears/python/PyCommentedCodeBear.py (hunk 1 above)
Comment (Makman2) [inline bears/python/PyCommentedCodeBear.py]: indeed this can be also made a tuple :+1:

File: bears/js/JSONFormatBear.py (hunk 2 above)
Comment (Makman2) [inline bears/js/JSONFormatBear.py]: shouldn't just removing `split_diff` suffice?

**Comment by Makman2** [comment]
reack 7b812bd, ef969d7

File: bears/js/JSONFormatBear.py [hunk 3]
```diff
@@ -48,8 +26,28 @@ def run(self,
...
+                file=filename)
+            return
+
+        corrected = json.dumps(json_content,
+                               sort_keys=json_sort,
+                               indent=tab_width,
+                               ensure_ascii=not keep_unicode).splitlines(True)
+        # Because of a bug in several python versions we have to correct
+        # whitespace here.
+        corrected = [line.rstrip(" \n") + "\n" for line in corrected]
```
Comment (sils1297) [inline bears/js/JSONFormatBear.py]: why not `tuple(` instead of `[`?

File: bears/js/JSONFormatBear.py [hunk 4]
```diff
@@ -48,8 +26,28 @@ def run(self,
...
+        diff = Diff.from_string_arrays(file, corrected)
+
+        if len(diff) > 0:
+            yield Result(self,
+                         "This file can be reformatted by sorting keys and "
+                         "following indentation.",
+                         affected_code=(diff.range(filename),),
```
Comment (sils1297) [inline bears/js/JSONFormatBear.py]: if the diff affects line 2 and 539 I don't want to get the whole range

**Comment by Makman2** [comment]
reack d6c98cc, d9d9b8a

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/392/commits/6191d999f7822f811db0440b1eb3f1a881af08ca https://github.com/coala-analyzer/coala-bears/pull/392/commits/9f9a5a6a1e9b7ab4cf71ef35dee831937dfead92

**Comment by sils1297** [comment]
coverage is full, checked manually

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**